src/rust/engine/process_execution/src/lib.rs

```diff
@@ -857,6 +859,47 @@ pub trait CommandRunner: Send + Sync + Debug {
   ) -> Result<FallibleProcessResultWithPlatform, ProcessError>;
 }
 
+/// A [`CommandRunner`] that can be used locally. This trait augments [`CommandRunner`] with
+/// methods needed by Nailgun and other users of local execution.
+#[async_trait]
```

Comment:
It doesn't look like this trait needs to be async. I'm also not sure it's carrying its weight?